core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/BlockletDataMapFactory.java

```diff
@@ -219,4 +225,27 @@ public DataMapMeta getMeta() {
     // TODO: pass SORT_COLUMNS into this class
     return null;
   }
+
+  @Override public SegmentProperties getSegmentProperties(String segmentId) throws IOException {
+    SegmentProperties segmentProperties = null;
```

Comment:
SegmentProperties need to be cached, it shouldn't read from disk always

Reply:
Done.